File: ArcherApp/app/src/main/java/me/mxcsyounes/archernotebook/viewmodels/ScoresViewModel.java
package me.mxcsyounes.archernotebook.viewmodels;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import java.util.List;

import me.mxcsyounes.archernotebook.database.entities.Score;
import me.mxcsyounes.archernotebook.repositories.ScoresRepository;

public class ScoresViewModel extends AndroidViewModel {

    private LiveData<List<Score>> mAllScores;
    private ScoresRepository mRepository;

    public ScoresViewModel(@NonNull Application application) {
        super(application);
        mRepository = new ScoresRepository(application);
        mAllScores = mRepository.getScores();
    }

    public LiveData<List<Score>> getAllScores() {
        return mAllScores;
    }

    public void deleteAllScores() {
        mRepository.deleteAllScores();
    }

    public void deleteScore(Score score) {
        mRepository.deleteScore(score);
    }

    public void updateScore(Score score) {
        mRepository.updateScore(score);
    }

    public void insertScore(Score score) {
        mRepository.insertScore(score);
    }

}

File: ArcherApp/app/src/main/java/me/mxcsyounes/archernotebook/viewmodels/ScoresViewModel.kt
package me.mxcsyounes.archernotebook.viewmodels

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData

import me.mxcsyounes.archernotebook.database.entities.Score
import me.mxcsyounes.archernotebook.repositories.ScoresRepository

class ScoresViewModel(application: Application) : AndroidViewModel(application) {

    val allScores: LiveData<MutableList<Score>>
    private val mRepository: ScoresRepository = ScoresRepository(application)

    init {
        allScores = mRepository.mScores
    }

    fun deleteAllScores() {
        mRepository.deleteAllScores()
    }

    fun deleteScore(score: Score) {
        mRepository.deleteScore(score)
    }

    fun updateScore(score: Score) {
        mRepository.updateScore(score)
    }

    fun insertScore(score: Score) {
        mRepository.insertScore(score)
    }

}
